napi_js_data.md

```diff
@@ -0,0 +1,760 @@
+In addition to types in that section, `napi_value_type` also can represent Functions and Objects with external data.
+
+### *napi_typedarray_type*
+
+#### Definition
+```
+enum napi_typedarray_type {
+  napi_int8_array,
+  napi_uint8_array,
+  napi_uint8_clamped_array,
+  napi_int16_array,
+  napi_uint16_array,
+  napi_int32_array,
+  napi_uint32_array,
+  napi_float32_array,
+  napi_float6_array4
+}
+```
+
+#### Description
+This represents the underlying binary scalar datatype of the TypedArray.
+Elements of this enum correspond to [Section 22.2 of the ECMAScript Language Specification](https://tc39.github.io/ecma262/#sec-typedarray-objects).
+
+## Functions
+
+### *napi_create_array*
+
+#### Signature
+```
+napi_status napi_create_array(napi_env env, napi_value* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the N-API API is invoked under
+- `[out] result`: A `napi_value` representing a JavaScript Array
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+#### Description
+This API returns a N-API value corresponding to a JavaScript Array type. 
+JavaScript arrays are described in [Section 22.1](https://tc39.github.io/ecma262/#sec-array-objects) of the ECMAScript Language Specification.
+
+### *napi_create_array_with_length*
+
+#### Signature
+```
+napi_status napi_create_array(napi_env env, size_t length, napi_value* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  length`: The initial length of the Array
+- `[out] result`: A `napi_value` representing a JavaScript Array
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+#### Description
+This API returns a N-API value corresponding to a JavaScript Array type. 
+The Array's length property is set to the passed in length parameter. 
+However, the underlying buffer is not guaranteed to be pre-allocated- that behavior
+is left to the underlying VM implementation. If you need the buffer to be 
+guaranteed, consider using `napi_create_external_arraybuffer`.
+
+JavaScript arrays are described in [Section 22.1](https://tc39.github.io/ecma262/#sec-array-objects) 
+of the ECMAScript Language Specification.
+
+### *napi_create_arraybuffer*
+
+#### Signature
+```
+napi_status napi_create_arraybuffer(napi_env env, size_t byte_length, void** data, napi_value* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  length`: The length in bytes of the array buffer to create
+- `[out] data`: Pointer to the underlying byte buffer of the ArrayBuffer
+- `[out] result`: A `napi_value` representing a JavaScript ArrayBuffer
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+#### Description
+This API returns a N-API value corresponding to a JavaScript ArrayBuffer.
+ArrayBuffers are used to represent fixed-length binary data buffers. 
+The ArrayBuffer allocated will have an underlying byte buffer whose size is determined 
+by the `length` parameter that's passed in. 
```

Comment:
Maybe mention how an arraybuffer is normally used as the backing buffer for a typedarray.